src/xmipp/libraries/parallel/mpi_angular_project_library.cpp

```diff
@@ -372,47 +372,47 @@ class ProgMpiAngularProjectLibrary: public ProgAngularProjectLibrary
             if(rank==0)
             {
 
-                MetaData  mySFin;
+                MetaDataVec  mySFin;
                 mySFin.read(output_file_root + "_angles.doc");
 #define ANGLESDOC
 #ifdef ANGLESDOC
 //       std::cerr << "DEBUG_ROB, output_file_root + angle.doc:"
 //    		     << output_file_root + "_angles.doc" << std::endl;
 #endif
-                MetaData  mySF;
+                MetaDataVec  mySF;
                 FileName fn_temp;
 
                 size_t myCounter = 0;
                 size_t id;
                 int ref;
-                for (int i = 0; i < std::ceil(360.0 / psi_sampling); ++i)
-                    //for (int i=0;i<=mysampling.no_redundant_sampling_points_angles.size()-1;i++)
-                    FOR_ALL_OBJECTS_IN_METADATA(mySFin)
-                {
-
-                    double x,y,z, rot, tilt, psi;
-                    mySFin.getValue(MDL_ANGLE_ROT,rot,__iter.objId);
-                    mySFin.getValue(MDL_ANGLE_TILT,tilt,__iter.objId);
-                    mySFin.getValue(MDL_ANGLE_PSI,psi,__iter.objId);
-                    mySFin.getValue(MDL_X,x,__iter.objId);
-                    mySFin.getValue(MDL_Y,y,__iter.objId);
-                    mySFin.getValue(MDL_Z,z,__iter.objId);
-                    mySFin.getValue(MDL_REF,ref,__iter.objId);
-
-                    //FIXME, do I have order?
-                    fn_temp.compose( ++myCounter,output_file);
-                    id = mySF.addObject();
-                    mySF.setValue(MDL_IMAGE,fn_temp, id);
-                    mySF.setValue(MDL_ENABLED,1, id);
-
-                    mySF.setValue(MDL_ANGLE_ROT,rot, id);
-                    mySF.setValue(MDL_ANGLE_TILT,tilt, id);
-                    mySF.setValue(MDL_ANGLE_PSI,psi + i * psi_sampling, id);
-                    mySF.setValue(MDL_X,x, id);
-                    mySF.setValue(MDL_Y,y, id);
-                    mySF.setValue(MDL_Z,z, id);
-                    mySF.setValue(MDL_SCALE,1.0,id);
-                    mySF.setValue(MDL_REF,ref,id);
+                for (int i = 0; i < std::ceil(360.0 / psi_sampling); ++i) {
```

Comment:
please move this outer loop after the line 399. Unless I missed something, reading from `mySFin` is independent on the `i`.

Reply:
Ok, done.